science/apbs/Portfile

```diff
@@ -20,64 +17,86 @@ long_description        APBS is a software package for the numerical \
                         electrostatic interactions between molecular \
                         solutes over a wide range of length scales.
 
-github.setup            Electrostatics apbs-pdb2pqr d4e78c62e6a07f92c26924318d83a159fa6af063
-version                 1.4.1
-homepage                http://www.poissonboltzmann.org
-worksrcdir              ${worksrcdir}/apbs
-revision                6
-
-# stealth update; remove for next version
-dist_subdir             ${name}/${version}_1
-checksums               rmd160  09ffa6a90c190eb4e584934519c8cbd6229fc7b8 \
-                        sha256  3c3e6914a1a69c27d54d9d9ea7ffd1072e3bc82209971289f0a8a7964fcdf82b \
-                        size    37120958
-
-depends_lib             port:maloc \
-                        port:readline
-
-# dependencies are not managed correctly within this code, so errors such as from -lapbs_geoflow might occur
-# how to solve: first 'make apbs_geoflow', then 'make all'
-use_parallel_build no
-
-configure.optflags      -O3 -ffast-math
-configure.args-append   -DENABLE_OPENMP:BOOL=OFF \
-                        -DENABLE_BEM=OFF \
-                        -DCMAKE_C_COMPILER_ARG1:STRING="-I${worksrcpath}/include ${configure.cflags}" \
-                        -DCMAKE_CXX_COMPILER_ARG1:STRING="-I${worksrcpath}/include ${configure.cxxflags} -stdlib=libc++"
-
-# otherwise obsolete subport installation will fail on patch phase and not get to pre-configure error about replacement
-if {${subport} eq ${name}} {
-    patchfiles              patch-CMakeLists.txt.diff patch-Eigen-include-path.diff patch-contrib-tabipb-CMakeLists.txt.diff patch-pbsolvercz.cpp.diff
-    if {[mpi_variant_isset]} {
-        patchfiles-append   patch-tests-apbs_tester.py.diff
+version                 3.4.1
+homepage                https://www.poissonboltzmann.org
+revision                0
+worksrcdir              ${workpath}/apbs-${version}
+
+extract.mkdir           no
+master_sites            https://github.com/Electrostatics/FETK/archive/refs/tags/ \
+                        https://github.com/Electrostatics/apbs/archive/refs/tags/
+distfiles               v${version}.tar.gz 1.9.3.tar.gz 
+checksums               1.9.3.tar.gz \
+                        rmd160  25cf57754a0b817e76e19b8de93617e554302661 \
+                        sha256  2ce7ab04cba4403f4208c3ecf1c81a0a18aae6a77d22da0ffa5f64c2da7c6f28 \
+                        size    27757348 \
+                        v${version}.tar.gz \
+                        rmd160  17de250c3887ff68f89a873c5a0be19f6ab38813 \
+                        sha256  644e6246fd37c9dd4172fc7be1763337082ab8584dbcd53e738c831b62c89bb2 \
+                        size    85806314
+
+# Test support requires python3 but the
+# building the apbs python module with swig
+# is unnecessary for apbs's use in pymol.
+platform darwin {
+   if {${os.major} < 13} {
+       depends_test      port:python312
     }
```

Comment:
```suggestion
    if {${os.major} < 19} {
        depends_test    port:python312
        set python ${prefix}/bin/python3.12
    } else {
        set python /usr/bin/python3
    }
```

Reply:
Corrected os.major with python3 system availability and expanded the conditional to set the python variable accordingly.